src/Package/Impl/Publishing/Handlers/RmdPublishHandler.cs

```diff
@@ -25,12 +30,26 @@ internal sealed class RmdPublishHandler : IMarkdownFlavorPublishHandler {
             return true;
         }
 
-        public string GetCommandLine(string inputFile, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
-            string format = GetDocumentTypeString(publishFormat);
-            string outputFile = Path.GetFileName(outputFilePath);
-            string outputFolder = Path.GetDirectoryName(outputFilePath).Replace('\\', '/');
-            // Run rmarkdown::render
-            return Invariant($"rmarkdown::render('{inputFile}', output_format='{format}', output_file='{outputFile}',  output_dir='{outputFolder}', encoding='cp{encoding.CodePage}')");
+        public async Task PublishAsync(IRSession session, ICoreShell coreShell,  IFileSystem fs, string inputFilePath, string outputFilePath, PublishFormat publishFormat, Encoding encoding) {
+            try {
+                using(var fts = new FileTransferSession(session, fs)) {
+                    var rmd = await fts.SendFileAsync(inputFilePath);
+                    var result = await RMarkdownRenderAsync(session, rmd, outputFilePath, publishFormat, encoding);
+                    await fts.FetchFileAsync(result, outputFilePath);
```

Comment:
To clarify, I think that there are two valid approaches here.

First one is when `RMarkdownRenderAsync` is strictly a remote-side affair: it takes a remote path for input file, another remote path for output file, runs markdown conversion, and that's that. If the caller needs to do it for local files, the onus is on them to use `FileTransferSession` to upload the input file before the conversion, and to download the output after. In this approach, `RMarkdownRenderAsync` doesn't do anything with blobs at all. This is more low-level, but provides more flexibility (e.g. input might be generated by some R code directly on remote FS, rather than uploaded).

The second approach is when `RMarkdownRenderAsync` works on the local side: i.e. it takes a local path for input file, and a local path for output file, and does all the magic that needs to happen to place the input on the remote FS, run conversion, and get the output back. So it uses `FileTransferSession` and blobs internally, but there's nothing in the API itself that indicates that. The returned task simply indicates completion, and is not a blob ID or anything like that - when it's complete, you just use the output file that you've specified. This is more high-level, and is easier to use, at the expense of flexibility.